geonode/people/views.py

```diff
@@ -145,6 +147,9 @@ class ProfileAutocomplete(autocomplete.Select2QuerySetView):
     def get_queryset(self):
         qs = get_user_model().objects.all().exclude(Q(username='AnonymousUser') | Q(is_active=False))
 
+        if self.request and self.request.user and not self.request.user.is_superuser:
```

Comment:
same as above is better to move the qs as an `else` condition to don't hit the db two times